src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -170,21 +171,65 @@ private JvmObjectReference CreateBroadcast(SparkContext sc, T value)
             bool encryptionEnabled = bool.Parse(
                 sc.GetConf().Get("spark.io.encryption.enabled", "false"));
 
+            var _pythonBroadcast = (JvmObjectReference)javaSparkContext.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.api.python.PythonRDD",
+                "setupBroadcast",
+                _path);
+
             if (encryptionEnabled)
             {
-                throw new NotImplementedException("Broadcast encryption is not supported yet.");
+                var pair = (JvmObjectReference[])_pythonBroadcast.Invoke("setupEncryptionServer");
+
+                using ISocketWrapper socket = SocketFactory.CreateSocket();
+                socket.Connect(
+                    IPAddress.Loopback,
+                    (int)pair[0].Invoke("intValue"),
+                    (string)pair[1].Invoke("toString"));
+                WriteInChunks(value, 8192, socket.OutputStream);
```

Comment:
can you create a constant?

Reply:
also you can just define this inside `WriteInChunks`

Reply:
Removed the need for `chunkSize` with latest code.